### PR TITLE
Ubuntu Action

### PR DESCRIPTION
**Pull request by K20shores**
This adds a github workflow that builds Cloud-J on ubuntu in debug and release mode with GNU

I had also tried to build on macOS and windows. The macOS build was failing for reasons unrelated to CloudJ and I couldn't figure out how to install netcdf-fortran on windows quickly. But I may put more time into either of those over the coming quarter. Hopefully this can help you identify build issues on different platforms as we start to integrate CloudJ with more NCAR software products

### CODE REVIEW

**Comment by lizziel** [comment]
Thanks @K20shores! I am tagging @yantosca who is also on the GEOS-Chem Support Team to review.

**Comment by lizziel** [comment]
@K20shores, heads up I have a cesm/dev branch in Cloud-J. Look at that one instead of main if you are browsing the code to assess needs for MUSICA. I need to do some work on the Cloud-J standalone data tables before bringing the more recent Cloud-J development work in.

File: .github/workflows/ubuntu.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        build_type: [Debug, Release]
+    env:
+      FC: gfortran
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```
Comment (yantosca) [inline .github/workflows/ubuntu.yml]: checkout@v4 is the most recent version now.

**Review by yantosca — CHANGES_REQUESTED** [review]
I think this is good to merge but I would use checkout@v4 instead of checkout@v3.

File: .github/workflows/ubuntu.yml (hunk 1 above)
Comment (K20shores) [reply .github/workflows/ubuntu.yml]: done

**Comment by K20shores** [comment]
@lizziel or @yantosca  would you mind checking to see if github actions are enabled for this repository?

**Comment by lizziel** [comment]
GitHub actions are enabled.

**Comment by K20shores** [comment]
@lizziel I'm not sure why the actions won't run, maybe it's because they aren't yet in this repository. You can see that they run and pass in a PR on my fork though: https://github.com/K20shores/Cloud-J/pull/1

**Comment by yantosca** [comment]
@K20shores maybe it won't run until the PR is merged into the main branch.  We could check this by pulling into a dev branch and then seeing if the PR can be started manually.

**Comment by lizziel** [comment]
Yes, I don't think the action will run unless it is merged in. I will go ahead and merge. If a fix is needed we can bring it in immediately.

**Review by yantosca — APPROVED** [review]
Thanks @K20shores!  Good to merge.

**Pull request merged**